tracer/src/Datadog.Tracer.Native/iast/module_info.cpp

```diff
@@ -0,0 +1,842 @@
+}
+
+std::vector<mdMethodDef> ModuleInfo::GetTypeMethodDefs(mdTypeDef typeDef)
+{
+    HCORENUM hCorEnum = nullptr;
+    std::vector<mdMethodDef> methods;
+    mdMethodDef enumeratedMethods[64];
+    ULONG enumCount;
+
+    while (this->_metadataImport->EnumMethods(&hCorEnum, typeDef, enumeratedMethods,
+                                             sizeof(enumeratedMethods) / sizeof(enumeratedMethods[0]),
+                                             &enumCount) == S_OK)
+    {
+        methods.insert(methods.end(), std::begin(enumeratedMethods), std::begin(enumeratedMethods) + enumCount);
+    }
+
+    this->_metadataImport->CloseEnum(hCorEnum);
+    return methods;
+}
+HRESULT ModuleInfo::GetTypeDef(const WSTRING& typeName, mdTypeDef* pTypeDef)
+{
+    HRESULT hr = S_OK;
+    auto parts = Split(typeName, WStr("+"));
+    if (parts.size() == 2)
+    {
+        mdTypeDef parentTypeDef;
+        hr = _metadataImport->FindTypeDefByName(parts[0].c_str(), 0, &parentTypeDef);
+        if (SUCCEEDED(hr))
+        {
+            hr = _metadataImport->FindTypeDefByName(parts[1].c_str(), parentTypeDef, pTypeDef);
+        }
+    }
+    else
+    {
+        hr = _metadataImport->FindTypeDefByName(typeName.c_str(), 0, pTypeDef);
+    }
+    return hr;
+}
+HRESULT ModuleInfo::GetMethodDef(mdTypeDef typeDef, const WSTRING& methodName, PCCOR_SIGNATURE pSignature,
+                                 ULONG nSignature, mdMethodDef* pMethodDef)
+{
+    auto methodInfo = GetMethod(typeDef, methodName, pSignature, nSignature);
+    if (methodInfo != NULL)
+    {
+        *pMethodDef = methodInfo->_id;
+    }
+    return methodInfo != NULL ? S_OK : E_INVALIDARG;
+}
+
+TypeInfo* ModuleInfo::GetTypeInfo(mdTypeDef typeDef)
+{
+    CSGUARD(_cs);
+    auto it = _types.find(typeDef);
+    if (it != _types.end())
+    {
+        return it->second;
+    }
+    TypeInfo* res = new TypeInfo(this, typeDef);
+    _types[typeDef] = res;
+    return res;
+}
+
+MemberRefInfo* ModuleInfo::GetMemberRefInfo(mdMemberRef token)
+{
+    auto typeFromToken = TypeFromToken(token);
+    if (typeFromToken == mdtMemberRef)
+    {
+        CSGUARD(_cs);
+        auto it = _members.find(token);
+        if (it != _members.end())
+        {
+            return it->second;
+        }
+        auto res = new MemberRefInfo(this, token);
+        _members[token] = res;
+        return res;
+    }
+    else if (typeFromToken == mdtMethodDef)
+    {
+        return GetMethodInfo(token);
+    }
+    else if (typeFromToken == mdtFieldDef)
+    {
+        return GetFieldInfo(token);
+    }
+    else if (typeFromToken == mdtMethodSpec)
+    {
+        return GetMethodSpec(token);
+    }
+    return NULL;
+}
+MethodInfo* ModuleInfo::GetMethodInfo(mdMethodDef methodDef)
+{
+    if (methodDef == 0x06000000)
+    {
+        return NULL; // Wrong methodDef
+    }
+    CSGUARD(_cs);
+    auto it = _methods.find(methodDef);
```

Comment:
same pattern here

Reply:
Fixed in 8d0ec8e7d5c0b4d08372d30f109a25db5fcc72f9